pkg/kubelet/cm/topologymanager/topology_manager_test.go

```diff
@@ -0,0 +1,664 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topologymanager
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask"
+	"k8s.io/kubernetes/pkg/kubelet/lifecycle"
+)
+
+func NewTestSocketMask(sockets ...int) socketmask.SocketMask {
+	s, _ := socketmask.NewSocketMask(sockets...)
+	return s
+}
+
+func NewTestSocketMaskFull() socketmask.SocketMask {
+	s, _ := socketmask.NewSocketMask()
+	s.Fill()
+	return s
+}
+
+func TestNewManager(t *testing.T) {
+	tcases := []struct {
+		name       string
+		policyType string
+	}{
+		{
+			name:       "Policy is set preferred",
+			policyType: "preferred",
+		},
+		{
+			name:       "Policy is set to strict",
+			policyType: "strict",
+		},
+		{
+			name:       "Policy is set to unknown",
+			policyType: "unknown",
+		},
+	}
+
+	for _, tc := range tcases {
+		mngr := NewManager(tc.policyType)
+
+		if _, ok := mngr.(Manager); !ok {
+			t.Errorf("result is not Manager type")
+		}
+	}
+}
+
+type mockHintProvider struct {
+	th []TopologyHint
+}
+
+func (m *mockHintProvider) GetTopologyHints(pod v1.Pod, container v1.Container) []TopologyHint {
+	return m.th
+}
+
+func TestGetAffinity(t *testing.T) {
+	tcases := []struct {
+		name          string
+		containerName string
+		podUID        string
+		expected      TopologyHint
+	}{
+		{
+			name:          "case1",
+			containerName: "nginx",
+			podUID:        "0aafa4c4-38e8-11e9-bcb1-a4bf01040474",
+			expected:      TopologyHint{},
+		},
+	}
+	for _, tc := range tcases {
+		mngr := manager{}
+		actual := mngr.GetAffinity(tc.podUID, tc.containerName)
+		if !reflect.DeepEqual(actual, tc.expected) {
+			t.Errorf("Expected Affinity in result to be %v, got %v", tc.expected, actual)
+		}
+	}
+}
+
+func TestCalculateAffinity(t *testing.T) {
```

Comment:
please add a test case where at least one hint provider returns an empty non-nil slice of TopologyHints (see other comment above about testing for empty slice)

Reply:
added below